urllib3/response.py

```diff
@@ -182,6 +192,38 @@ def tell(self):
         """
         return self._fp_bytes_read
 
+    def _init_length(self, request_method):
+        """
+        Set initial length value for Response content.
+
+        Logic extracted from httplib.py in the standard Python library.
+        """
+        length = self.headers.get('content-length')
+        if length is not None and not self.chunked:
+            try:
+                length = int(length)
+            except ValueError:
+                length = None
+            else:
+                if length < 0:
+                    length = None
+        else:
+            length = None
+
+        # Convert status to int for comparison
+        # In some cases, httplib returns a status of "_UNKNOWN"
+        try:
+            status = int(self.status)
+        except:
+            status = 0
+
+        # Check for responses that shouldn't include a body
+        if (status in (204, 304) or 100 <= status < 200
+           or request_method == 'HEAD'):
+            length = 0
+
+        return length
```

Comment:
@Lukasa are you seeing the more aggressive content-length validation as part of a urllib3 2.0/#someday feature, or something that would be integrated before a major (1.20+). It seems breaking, especially if people are using this for testing purposes. I wanted to record a few of the changes that were discussed here but passed on for later PRs.